hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -73,6 +78,8 @@
   private long bytesFromReadAhead; // bytes read from readAhead; for testing
   private long bytesFromRemoteRead; // bytes read remotely; for testing
 
+  private IOStatistics ioStatistics;
```

Comment:
final?

Reply:
Could be null too, depending on the stream statistics instance(Which we can be either be null or not, using the AbfsInputStreamContext class).